src/prefect/utilities/logging.py

```diff
@@ -33,16 +33,44 @@
 )
 
 
+class PrefectFormatter(logging.Formatter):
+    """
+    Custom formatter implementing timezone localization.
+    Creates a logger with a `StreamHandler` that has level and formatting
+    set from `prefect.config`.
+
+    Args:
+        - display_tz (str): Typically set from `prefect.config`, used by `formatTime`
+        method for display in log records.
+    """
+
+    def __init__(self, *args, display_tz: str = "UTC", **kwargs) -> None:
+        self.display_tz = display_tz
+        super().__init__(*args, **kwargs)
+
+    def converter(self, timestamp: float) -> pendulum.DateTime:
+        return pendulum.from_timestamp(timestamp, tz=self.display_tz)
+
+    def formatTime(self, record, datefmt=None) -> str:
```

Comment:
A docstring may be nice explaining how this specifically differs from the parent class' implementation. For reference for other reviewers

```python
        ct = self.converter(record.created)
        if datefmt:
            s = time.strftime(datefmt, ct)
        else:
            t = time.strftime(self.default_time_format, ct)
            s = self.default_msec_format % (t, record.msecs)
        return s
```